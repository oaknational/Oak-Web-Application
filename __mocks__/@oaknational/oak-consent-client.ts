const consentClientMock = jest.requireActual("@oaknational/oak-consent-client");

/**
 * Provides a mock implementation of OakConsentClient so that tests
 * do not try to hit the network
 */
class MockOakConsentClient {
  getState() {
    return {
      policyConsents: [],
      requireInteraction: false,
    };
  }
  logConsents() {
    return Promise.resolve();
  }
  getConsent() {
    return "pending";
  }
  onStateChange() {
    return () => {};
  }
}

module.exports = {
  ...consentClientMock,
  OakConsentClient: MockOakConsentClient,
};
