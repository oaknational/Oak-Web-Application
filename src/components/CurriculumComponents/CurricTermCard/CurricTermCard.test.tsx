import { OakBox } from "@oaknational/oak-components";

import { CurricTermCard } from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("CurricTermCard", () => {
  test("snapshot", () => {
    const { baseElement, getByTestId } = renderWithTheme(
      <CurricTermCard
        title={"Yeat 1"}
        coveredNumberOfLessons={25}
        totalNumberOfLessons={30}
      >
        <OakBox data-testid="content">TESTING</OakBox>
      </CurricTermCard>,
    );
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toHaveTextContent(`25/30 lessons scheduled`);
    expect(getByTestId("content")).toBeDefined();
  });
});
