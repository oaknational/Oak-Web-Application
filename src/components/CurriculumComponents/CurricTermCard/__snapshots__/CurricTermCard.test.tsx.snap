// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurricTermCard snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-kAyceB sSUCJ moWty"
    >
      <div
        class="sc-fqkvVR fgBWrW"
      >
        <h5
          class="sc-kpDqfm evKItP"
        >
          Yeat 1
        </h5>
        <div
          class="sc-fqkvVR sc-kAyceB fgBWrW gCgXGf"
        >
          <span
            class="sc-dAlyuH gAWMRU"
          >
            25
            /
            30
             lessons scheduled
          </span>
          <div
            class="sc-fqkvVR erqloa"
          >
            <img
              alt=""
              class="sc-dcJsrY cDKKvZ"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-fqkvVR fgBWrW"
        data-testid="content"
      >
        TESTING
      </div>
    </div>
  </div>
</body>
`;
